### PR TITLE
HWKMETRICS-630 Alert quietCount option

### PR DESCRIPTION
**Pull request by jshaughn**
@lucasponce  This feature may be OK as is but we should keep it open until things are clear for hwkalerts-240.

### CODE REVIEW

**Comment by lucasponce** [comment]
As commented in the JIRA, I am skeptical about this option, as basically it introduces a sort of dampening effect.
Perhaps I don't have enough examples/use cases for it.
Could you ellaborate more use cases to discuss it ?

**Comment by jshaughn** [comment]
@lucasponce, I responded in HWKALERTS-240.  Hopefully that post explains well why I think this is still a useful option for the alerter, even if we don't use the alerter as the solution for that issue.

Please review as just an additional alerter feature.

**Comment by jshaughn** [comment]
retest this please

**Comment by jshaughn** [comment]
@lucasponce Unless you have an issue, I think this can be merged.

File: alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/ConditionManager.java [hunk 1]
```diff
@@ -489,15 +495,17 @@ private void evaluateEach(String target, Map<String, Map<String, List<? extends
...
             Map<String, String> preparedCondition = null;
             try {
                 preparedCondition = evaluator.prepare(queryResults);
             } catch (Exception e) {
                 log.debugf("Could not prepare evaluation, Skipping due to [%s]", e.getMessage());
             }
 
-            if (evaluator.evaluate()) {
+            boolean evalResult = evaluator.evaluate();
+            if (!isQuiet("_ALL_", expression.getQuietCount(), evalResult)) {
```
Comment (lucasponce) [inline alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/ConditionManager.java]: Minor one: I wonder if instead to treat "_ALL_" as a metricName, if it is not better just to pass a boolean flag to indicate all/each.

**Comment by lucasponce** [comment]
@jshaughn sorry for the delay, overall LGTM, I have a minor comment but no blocker.
This feature can get merged.

File: alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/ConditionManager.java (hunk 1 above)
Comment (jshaughn) [inline alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/ConditionManager.java]: I see your point, but I guess for now I'd prefer to leave things unchanged.  By using a special metricName the underlying mechanism can just use it as a Map key and not have to know it's something special.

**Comment by stefannegrea** [comment]
Thank you @jshaughn!

**Pull request merged**